numpy/lib/function_base.py

```diff
@@ -5010,3 +5011,76 @@ def append(arr, values, axis=None):
         values = ravel(values)
         axis = arr.ndim-1
     return concatenate((arr, values), axis=axis)
+
+
+def scale(arr, min=0, max=1, axis=None, out=None):
+    """Linearly scale non-NaN values in array to interval [min, max].
+
+    .. versionadded:: 1.12.0
+
+    Parameters
+    ----------
+    arr : array_like
+        Input array.
+    min : scalar or array_like, optional
+        The minimum value in the output array (default: 0). If array-like,
+        its shape must match that of projection of `arr` on `axis`.
+    max : scalar or array_like, optional
+        The maximum value in the output array (default: 1). If array-like,
+        its shape must match that of projection of `arr` on `axis`.
+    axis : int, optional
+        Axis along which to scale `arr`. If `None`, scaling is done over
+        the flattened array.
+    out : array_like, optional
+        Array of the same shape as `arr`, in which to store the results.
+
+    Returns
+    -------
+    scaled_array : ndarray
+        Output array with the same shape as `arr` and all non-NaN values
+        scaled to interval [min, max].
+
+    Raises
+    ------
+    ValueError
+        When ``min > max`` or when values of ``min``, ``max``, and ``axis``
+        parameters are incompatible.
+
+    Examples
+    --------
+    >>> a = np.arange(6).reshape((2, 3))
+    >>> np.scale(a)
+    array([[ 0. ,  0.2,  0.4],
+           [ 0.6,  0.8,  1. ]])
+    >>> np.scale(a, -1, 1, axis=1)
+    array([[-1.,  0.,  1.],
+           [-1.,  0.,  1.]])
+    >>> np.scale(a, [0, 1, 0], [1, 1, 1], axis=0)
+    array([[ 0.,  1.,  0.],
+           [ 1.,  1.,  1.]])
+    """
+    arr, min, max = asanyarray(arr), asanyarray(min), asanyarray(max)
+
+    if np.any(min > max):
+        raise ValueError('min must be less than max')
+    if axis is None and (min.ndim or max.ndim):
+        raise ValueError('min and max must be scalar when axis is None')
+    if axis is not None:
+        valid_shapes = ((), arr.shape[:axis] + arr.shape[axis + 1:])
+        if min.shape not in valid_shapes or max.shape not in valid_shapes:
+            raise ValueError('Invalid shape of min / max. Shapes should be: {}'
+                             .format(valid_shapes[-1]))
+
+    if out is None:
+        out = np.empty_like(arr, dtype=float)
+
+    mins, maxs = np.nanmin(arr, axis), np.nanmax(arr, axis)
+
+    if axis is None:
+        out[...] = (arr - mins) / (maxs - mins) * (max - min) + min
+    else:
+        arr = np.rollaxis(arr, axis)
+        out[...] = np.rollaxis(
```

Comment:
Thanks. I agree it's much better. Have I gotten the code correctly and concise enough?